lib/net/ssh/authentication/agent.rb

```diff
@@ -139,6 +147,37 @@ def sign(key, data)
       return reply.read_string
     end
 
+    # Adds the private key with comment to the agent.
+    # If opts[:lifetime] is given, the key will automatically be removed after
+    # opts[:lifetime] seconds.
+    # If opts[:confirm] is true, confirmation will be required for each agent
+    # signing operation.
+    def add_identity(priv_key, comment, opts={})
```

Comment:
instead of opt, we could use kw arguments? what do you think? We now require ruby 2.0.